### PR TITLE
Allow SQL interface to be enabled without "experimental" compat flag.

### PR DESCRIPTION
**Pull request by kentonv**
Instead, it can be enabled in the workerd config.

In production, it'll be enabled implicitly if the new SQLite-based storage engine is in use. But `workerd` running locally always uses SQLite, even when testing code intended to run against non-SQLite namespaces in production, so we need a config option for workerd.

**Also enable PITR and abort() APIs without "experimental".**

Point-in-time recovery is only supported for SQLite-backed DOs in production; PITR methods will throw for non-SQLite namespaces.

abort() will work for all DOs. I think we've generally agreed this should be made public, but it's especially needed with the PITR APIs since you need a way to restart the object immediately after calling `onNextSessionRestoreBookmark()`.

### CODE REVIEW

**Review by justin-mp — APPROVED** [review]

File: src/workerd/server/workerd.capnp [hunk 1]
```diff
@@ -579,6 +579,14 @@ struct Worker {
     # pinned to memory forever, so we provide this flag to change the default behavior.
     #
     # Note that this is only supported in Workerd; production Durable Objects cannot toggle eviction.
+
+    enableSql @4 :Bool;
```
[the remark this answers is not in the document]
Comment (kentonv) [reply src/workerd/server/workerd.capnp:583]: This field isn't a member of a union. The union ended on line 574.

> Also, it appears we are manually copying this file over here(workers-sdk)? 😕

Manual copying is a pretty normal way to share schemas. It's not a big deal, you just overwrite the copy with the updated version if and when you want to change workers-sdk to use the new field.

**Comment by kentonv** [comment]
After some discussion I think I'm going to update this PR so that instead of making `storage.sql` "optional", it will always be present, but its methods will throw exceptions if the DO is not backed by SQL.

**Comment by kentonv** [comment]
Updated as described: Now `storage.sql` is always there, but the methods throw exceptions if the object isn't SQLite-backed.

File: src/workerd/server/workerd.capnp (hunk 1 above)
[the remark this answers is not in the document]
Comment (joshthoward) [reply src/workerd/server/workerd.capnp:583]: My bad!

**Review by geelen — APPROVED** [review]

**Pull request merged**